File: Kata 6/Array.diff/Arraydiff.py
def array_diff(a, b):
    return [ x for x in a if all (x !=y for y in b)]
	
def array_diff(a, b):
    return [ x for x in a if x not in b]
	
def array_diff(a, b):   
    return filter(lambda i: i not in b, a)
    
def array_diff2(a, b):
    a.sort()
    b.sort()
    p1=0
    p2=0
    while(p2<len(b) and p1<len(a)):
        if(a[p1]==b[p2]):
            a.remove(a[p1])
        elif a[p1]<b[p2]:
            p1+=1
        elif a[p1]>b[p2]:
            p2+=1
    return a    
	

File: Kata 6/Array.diff/Arraydiff.js
function array_diff(a, b) {
    var result = [];
    for (var i = 0; i<a.length ;i++){
        var bappear = false;
        for (var j = 0 ; j < b.length ; j++){
            if (a[i]== b[j]){
              bappear = true;
              break;
            }
        }
        if (!bappear)
          result.push(a[i]);
        
    }
    return result;
    
}



function array_diff_filter(a, b) {

    //lambda
    //return a.filter(x => !b.includes(x))
    
    return a.filter(function(x){b.indexOf(x)==-1;});   
    
}


/*
1. b.includes(x)
2. b.indexOf(x)
3. b = new Set(b) ,b.has(v)


















*/












